projects/blenderbot2/agents/blenderbot2.py

```diff
@@ -898,6 +905,39 @@ def compute_loss(
         else:
             return loss
 
+    def _clean_text(self, txt):
+        cleaned_txt = re.sub(r'_[\S]*unsafe_*', '', txt, flags=re.IGNORECASE)
+        return cleaned_txt.strip()
+
+    def self_observe(self, self_message: Message) -> None:
+        """
+        Observe one's own utterance.
+        Override TorchAgent.self_observe with the optional cleaned text
+
+        :param self_message:
+            The message corresponding to the output from batch_act.
+        """
+
+        if (
+            self.opt('clean_reply', False)
```

Comment:
So it returns here and doesn't go to cleaning the text if `clean_reply` is `True`?

Reply:
I'd love to ALWAYS clean the reply and I think it's the right thing to do?

Reply:
I'd love to ALWAYS clean the reply and I think it's the right thing to do?